src/Microsoft.AspNetCore.Routing/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -303,6 +304,21 @@ public override int GetDestination(HttpContext httpContext)
 
                 return _destinations.TryGetValue(httpMethod, out destination) ? destination : _exitDestination;
             }
+
+            public void ToGraph(string label, TextWriter writer)
+            {
+                writer.WriteLine($"{label} -> {_exitDestination} [label=\"HTTP: {JumpTable.FallbackEdgeLabel}\"]");
+
+                foreach (var kvp in _destinations)
+                {
+                    writer.WriteLine($"{label} -> {kvp.Value} [label=\"HTTP: {kvp.Key}\"]");
+                }
+
+                foreach (var kvp in _corsPreflightDestinations)
+                {
+                    writer.WriteLine($"{label} -> {kvp.Value} [label=\"CORS Preflight: {kvp.Key}\"]");
+                }
+            }
```

Comment:
What will be displayed for custom policies?

Reply:
Nothing right now, we can't deeply introspect custom jump tables. That would require a public API for this. If we're convinced its valuable, I'm not opposed to a pubternal API for this.

Reply:
Another idea that I have here is to make this part of the DFA builder, which might not require that.

Reply:
I'm going to try doing this in the DFA matcher builder so we can capture custom policies as well.